daemon/containerd/soft_delete.go

```diff
@@ -17,23 +17,18 @@ const imageNameDanglingPrefix = "moby-dangling@"
 // softImageDelete deletes the image, making sure that there are other images
 // that reference the content of the deleted image.
 // If no other image exists, a dangling one is created.
-func (i *ImageService) softImageDelete(ctx context.Context, img containerdimages.Image) error {
-	is := i.client.ImageService()
-
-	// If the image already exists, persist it as dangling image
-	// but only if no other image has the same target.
-	dgst := img.Target.Digest.String()
-	imgs, err := is.List(ctx, "target.digest=="+dgst)
-	if err != nil {
-		return errdefs.System(errors.Wrapf(err, "failed to check if there are images targeting digest %s", dgst))
-	}
-
+func (i *ImageService) softImageDelete(ctx context.Context, img *containerdimages.Image, imgs []containerdimages.Image) error {
 	// From this point explicitly ignore the passed context
 	// and don't allow to interrupt operation in the middle.
 
+	// If no explicit image to delete, just return and keep remaining refereneces
+	if img == nil {
+		return nil
+	}
+
```

Comment:
If nil img doesn't do anything, maybe we should just accept non nilable `containerdimages.Image`?